packages/model-viewer/src/features/scene-graph/model.ts

```diff
@@ -57,7 +290,23 @@ export class Model implements ModelInterface {
    *
    * TODO(#1003): How do we handle non-active scenes?
    */
-  get materials(): Array<Material> {
+  get materials(): Material[] {
     return this[$materials];
   }
+
+  async[$switchVariant](variantName: string) {
```

Comment:
So, one thing I don't love about the variant spec is that the initially loaded configuration is not necessarily the same materials as any one of the named variants. I'd say that's a poor practice, but we might want to think about how one might get back to the "default" state. That can wait for a follow-on if it's not simple.

Reply:
Ahh, actually the previous implementation did that but we avoid it now because we are mapping variants to _model-viewer_ materials, after they list has been created, so the "default" material loaded will be always reference a material that was previously loaded.

Reply:
Okay, but say I selected a variant and now I want to go back. Is there a `variantName` I can choose that will take me to the original configuration?

Reply:
I think I see, the 'original' should be one of the variants yes? So you would have to know which one was the default then go back to that. If not, then I'd call that an authoring mistake, also the spec is letting you down. We could add a `revertToDefaultVariant` to help, but yeah the variant spec has some edge cases, it feels very bolted on.

Reply:
No, that's exactly the problem. I think it would be a good idea, but sadly the spec does not require any of the variants to equal the non-extension material setup. The extension mappings are completely independent of the non-extension materials, which are what you get when you first load it. I think a simple solution might just be to accept `null` and put it back then. But I'd guess you may have to do some work when creating your mappings to save this as a special state.

Reply:
Sounds like an implied 'extra' variant, I think that's a small change to just mark the 'initial' material, but I'd want to be very specific about switch back to it if it's the 'implied' material though, to avoid false positives, where we just end up switching back to the initial material because we got some noise that couldn't quite be parsed.